### PR TITLE
chore: update rust edition

### PR DESCRIPTION
**Pull request by flavio**
Move to the 2021 edition

### CODE REVIEW

**Review by jvanz — APPROVED** [review]

**Comment by flavio** [comment]
@kubewarden/kubewarden-developers: please take a look at this PR. If we upgrade to the latest Rust edition we will be forced to use `make check`.

Take a look at this commit message: https://github.com/kubewarden/policy-evaluator/pull/216/commits/e45a9b581fadee9a8d9fb5bf36f6285e32d0e850

File: Makefile [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-	cargo fmt --all -- --check
+	K8S_OPENAPI_ENABLED_VERSION=1.24 cargo fmt --all -- --check
 
 .PHONY: lint
 lint:
-	cargo clippy --workspace -- -D warnings
+	K8S_OPENAPI_ENABLED_VERSION=1.24 cargo clippy --workspace -- -D warnings
 
 .PHONY: test
 test: fmt lint
```
Comment (jvanz) [inline Makefile:14]: This is not a requirement, but I personally prefer setting values in variables. Thus, it's easy to change them with no code change. Something like this:

```
+API_VERSION?=1.24
+
 .PHONY: build
 build:
-       K8S_OPENAPI_ENABLED_VERSION=1.24 cargo build --release
+       K8S_OPENAPI_ENABLED_VERSION=$(API_VERSION) cargo build --release
 
 .PHONY: fmt
 fmt:
-       K8S_OPENAPI_ENABLED_VERSION=1.24 cargo fmt --all -- --check
+       K8S_OPENAPI_ENABLED_VERSION=$(API_VERSION) cargo fmt --all -- --check
 
 .PHONY: lint
 lint:
-       K8S_OPENAPI_ENABLED_VERSION=1.24 cargo clippy --workspace -- -D warnings
+       K8S_OPENAPI_ENABLED_VERSION=$(API_VERSION) cargo clippy --workspace -- -D warnings

```

**Review by jvanz — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (flavio) [reply Makefile:14]: this makes a lot of sense. fixed

**Review by viccuad — APPROVED** [review]
LGTM

**Pull request merged**